vasiliy/src/main/java/data_structures/MyLinkedList.java

```diff
@@ -1,6 +1,207 @@
 package data_structures;
 
-public class MyLinkedList {
+public class MyLinkedList implements MyList {
     private Node head;
     private Node tail;
+    private int size;
+
+    public MyLinkedList(Object[] array) {
+        for (Object obj : array) {
+            this.add(obj);
+        }
+    }
+
+    private boolean checkIndex(int index) {
+        return index >= 0 && index < size;
+    }
+
+    @Override
+    public void clear() {
+        tail = null;
+        head = null;
+        size = 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    @Override
+    public boolean add(Object o) {
+        if (isEmpty()) {
+            head = new Node(o);
+            tail = head;
+        } else {
+            Node node = new Node(o, null, tail);
+            tail.setNext(node);
+            tail = node;
+        }
+        size++;
+        return true;
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return checkIndex(indexOf(o));
+    }
+
+    @Override
+    public boolean add(int index, Object o) {
+        if (!checkIndex(index)) {
```

Comment:
You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }